crates/hyperswitch_connectors/src/connectors/novalnet/transformers.rs

```diff
@@ -125,99 +321,687 @@ impl<F, T> TryFrom<ResponseRouterData<F, NovalnetPaymentsResponse, T, PaymentsRe
+            let novalnet_redirection_response =
+                serde_urlencoded::from_str::<NovolnetRedirectionResponse>(encoded_data.as_str())
+                    .change_context(errors::ConnectorError::ResponseDeserializationFailed)?;
+            NovalnetSyncTransaction {
+                tid: novalnet_redirection_response.tid,
+            }
+        } else {
+            NovalnetSyncTransaction {
+                tid: item
+                    .request
+                    .get_connector_transaction_id()
+                    .change_context(errors::ConnectorError::MissingConnectorTransactionID)?,
+            }
+        };
+
+        let custom = NovalnetCustom {
+            lang: "EN".to_string(),
+        };
+        Ok(Self {
+            transaction,
+            custom,
+        })
+    }
+}
+
+impl<F>
+    TryFrom<ResponseRouterData<F, NovalnetPSyncResponse, PaymentsSyncData, PaymentsResponseData>>
+    for RouterData<F, PaymentsSyncData, PaymentsResponseData>
+{
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(
+        item: ResponseRouterData<F, NovalnetPSyncResponse, PaymentsSyncData, PaymentsResponseData>,
+    ) -> Result<Self, Self::Error> {
+        match item.response.result.status {
+            NovalnetAPIStatus::SUCCESS => {
+                let transaction_id = item
+                    .response
+                    .transaction
+                    .clone()
+                    .map(|data| data.tid.to_string());
+                let transaction_status = item
+                    .response
+                    .transaction
+                    .map(|transaction_data| transaction_data.status)
+                    .unwrap_or(NovalnetTransactionStatus::PENDING);
+
+                Ok(Self {
+                    status: common_enums::AttemptStatus::from(transaction_status),
+                    response: Ok(PaymentsResponseData::TransactionResponse {
+                        resource_id: transaction_id
+                            .clone()
+                            .map(ResponseId::ConnectorTransactionId)
+                            .unwrap_or(ResponseId::NoResponseId),
+                        redirection_data: None,
+                        mandate_reference: None,
+                        connector_metadata: None,
+                        network_txn_id: None,
+                        connector_response_reference_id: transaction_id.clone(),
+                        incremental_authorization_allowed: None,
+                        charge_id: None,
+                    }),
+                    ..item.data
+                })
+            }
+            NovalnetAPIStatus::FAILURE => {
+                let response = Err(get_error_response(item.response.result, item.http_code));
+                Ok(Self {
+                    response,
+                    status: enums::AttemptStatus::Failure,
+                    ..item.data
+                })
```

Comment:
Has this block been tested with their APIs?

Reply:
no not able to reproduce locally with hyperswitch, but I have tested this case in postman with  Novalnet API directly